core/vm/jump_table.go

```diff
@@ -57,21 +58,39 @@ var (
 // JumpTable contains the EVM opcodes supported at a given fork.
 type JumpTable [256]*operation
 
+func validate(jt JumpTable) JumpTable {
+	for i, op := range jt {
+		if op == nil {
+			panic(fmt.Sprintf("op 0x%x is not set", i))
+		}
+		// The interpreter has an assumption that if the memorySize function is
```

Comment:
Considering they always go together, it would be better to merge these into single function that outputs new memory size and error.

Reply:
Nitpick: they don't "always go together". If dynamic memory, then dynamic gas, but not always the other way around. For example the `EXP` or `SSTORE` and `SELFDESTRUCT` doesn't expand memory but have a dynamic cost. 

But yes, maybe something could be changed in the strucutre at a higher level.

Reply:
Yes, what I meant in this case they should return unchanged memory size or 0.
But in total this reduces helper calls from 2 to 1 and eliminates the need for `validate()`.

Reply:
> eliminates the need for validate().

Afaik, the validate is only performed during geth bootup, when the global vars are instantiated. Unless I misunderstood something.

Reply:
I agree that it can be reworked, but I don't want to do it in this PR. I think merging memory funcs and gas funcs is going to be a very large change.

Reply:
> Afaik, the validate is only performed during geth bootup, when the global vars are instantiated. Unless I misunderstood something.

Correct. But my comment was about the design. Not needing `validate()` is better design.

> I agree that it can be reworked, but I don't want to do it in this PR.

Make sense.

Reply:
I am doing this refactoring in https://github.com/ethereum/go-ethereum/pull/24113